### ISSUE TITLE
Double boundary points 

### ISSUE THREAD

**Issue by BureauNL** [lululxvi/deepxde#66]
Dear @lululxvi 

I am trying to train a network using adaptive collocation points on a 2D space. I noticed something:
It seems that the boundary points are added twice to the list of collocation points; once as boundary points (Dirichlet in this case) and once as part of the PDE. 

![trainingset](https://user-images.githubusercontent.com/66969838/84707388-e3b07380-af5e-11ea-86ba-e5cef40d0b84.png)

Since the PDE is only defined in the interior, is there a neural network related reason for why these points are added? The way it is now, it is trying to solve both, resulting in a bigger error.

Thanks in advance.

Code:
````from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from mpl_toolkits.mplot3d import Axes3D

import numpy as np
import tensorflow as tf
import scipy as sc
import importlib as importlib
import matplotlib.pyplot as plt

import deepxde as dde
importlib.reload(dde.model)

def main():
    def pde(x, y):
        dy_x = tf.gradients(y, x)[0]
        dy_x, dy_y = dy_x[:, 0:1], dy_x[:, 1:]
        dy_xx = tf.gradients(dy_x, x)[0][:, 0:1]
        dy_yy = tf.gradients(dy_y, x)[0][:, 1:]
        return -dy_xx - dy_yy

    def boundary(_, on_boundary):
        return on_boundary

    def func(x):
        return np.zeros([len(x), 1])

    gridnumber = 3
    boundarypoints = 4*(gridnumber-1)
    initialinteriorpoints = gridnumber**2 - boundarypoints

    #initial grid, based on 1x1 grid in positive corner
    gridanchors = np.zeros([gridnumber*gridnumber-boundarypoints,2])
    for i in range (gridanchors.shape[0]):
        gridanchors[i,0] = (i % (gridnumber-2) + 1)/(gridnumber-1)
        gridanchors[i,1] = (i//(gridnumber-2) + 1)/(gridnumber-1)

    #Initial model
    geom = dde.geometry.Rectangle([0,0],[1,1])
    bc = dde.DirichletBC(geom, func, boundary)
    data = dde.data.PDE(geom, pde, bc, num_domain=0, num_boundary=boundarypoints, num_test=169, anchors=gridanchors)
    net = dde.maps.FNN([2] + [50] * 4 + [1], "tanh", "Glorot uniform")
    model = dde.Model(data, net)

    model.compile("adam", lr=0.001)
    losshistory, train_state = model.train(epochs=3000, display_every=300)
    dde.saveplot(losshistory, train_state, issave=True, isplot=True)
    
    datapoints = np.genfromtxt("train.dat")
    print(datapoints)    

if __name__ == "__main__":
    main()
````

**Comment by lululxvi** [comment]
Most PDEs are satisfied also on the boundary points. Also, in practice, it **may** improve the accuracy.
